### PR TITLE
Build in docker

### LINKED ISSUE #1: Create build container image
Initial solution would be based on SDL2 library. 

- Create docker image that can build C++ application linked with SDL2 lib:
1. for *nix platform (Ubuntu 22.04)
2. for win* platform (Windows 10\11)

- Push docker image to docker-hub
- Add link to image in readme.md

### PR DESCRIPTION
**Pull request by Drone29**
Revised structure, 3rd party packages moved to ImageDockerfile and removed from source tree.
SDL, SDL_Image,... fixed release versions used from now on
.cmake config modules renamed and moved to config/ folder
src/ now set to contain both .cpp and .h files
Added GTest submodule with fixed version v1.14.0 under test/utest
closes #1

### CODE REVIEW

File: .gitmodules
Comment (MikhailBerezhanov) [inline .gitmodules:1]: can be removed

File: assets/images/sdl_logo.bmp
Comment (MikhailBerezhanov) [inline assets/images/sdl_logo.bmp:1]: Needed ?

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,97 @@
+cmake_minimum_required(VERSION 3.10.0)
+
+project(test_game VERSION 0.1.0 LANGUAGES C CXX)
+
+# Set the C++ standard
+set(CMAKE_CXX_STANDARD 14)
```
Comment (MikhailBerezhanov) [inline CMakeLists.txt:6]: Lets use 20

File: ImageDockerfile
Comment (MikhailBerezhanov) [inline ImageDockerfile:1]: Better to rename as `Dockerfile` to let syntax detection in IDEs  and place in `builder/` subdir

File: ImageDockerfile [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+RUN rm -rf ./*
+
+# Install for Windows (MinGW)
+RUN cmake .. -DCMAKE_C_COMPILER=x86_64-w64-mingw32-gcc \
+    -DCMAKE_RC_COMPILER=x86_64-w64-mingw32-windres \
+    -DCMAKE_SYSTEM_NAME=Windows \
+    -DCMAKE_INSTALL_PREFIX=${SDL_PREFIX_WINDOWS}
+
+RUN cmake --build .
```
Comment (MikhailBerezhanov) [inline ImageDockerfile:144]: `--config Release` ?

**Comment by MikhailBerezhanov** [comment]
why `config/` dir for cmake files ? why not `cmake/` ?

**Review by MikhailBerezhanov — CHANGES_REQUESTED** [review]
Lets rework directories naming and structure a bit more .

File: assets/images/sdl_logo.bmp
Comment (Drone29) [reply assets/images/sdl_logo.bmp:1]: for testing purposes

File: .gitmodules
Comment (Drone29) [reply .gitmodules:1]: +

File: CMakeLists.txt (hunk 1 above)
Comment (Drone29) [reply CMakeLists.txt:6]: not gonna work. c++17 tops

**Review by MikhailBerezhanov — APPROVED** [review]

**Pull request merged**